backend/src/connectors/peer/base.ts

```diff
@@ -56,4 +70,17 @@ export abstract class BasePeerConnector {
    * Fetch the peer's system status
    */
   abstract getPeerStatus(): Promise<SystemStatus>
+
+  abstract queryModels(opts, user: UserInterface): Promise<Array<ModelSearchResult>>
+
+  /**
+   * Namespace cache by user
+   *
+   * @param user to extract the DN from
+   * @param key to use as the remainder of the key (such as a query string)
+   * @returns a user-specific cache key
+   */
+  buildCacheKey(user: UserInterface, key: string) {
+    return `${user.dn}+${key}`
```

Comment:
I'm not super keen on using a raw `user.dn` as that could include unexpected characters in it depending on things like region and language set. I would be happier if we base-64 encode it first e.g. `Buffer.from(user.dn).toString('base64')`. This would still be user specific.

Reply:
Also, this is only per-user and not also per-peer, so caches with multiple peers night have collisions.
```suggestion
    return `${this.id}:${Buffer.from(user.dn).toString('base64')}+${key}`
```